### ISSUE TITLE
Terminal opens wrong path if focus is on tree-view

### ISSUE THREAD

**Issue by rm-code** [tkw1536/atomerinal#2]
Hello @tkw1536,

there seems to be a problem with atomerinal if the tree view contains multiple folders:

![screenshot 2016-05-16 14 08 48](https://cloud.githubusercontent.com/assets/11627131/15289048/bcfeab4e-1b6f-11e6-93b9-b39c15ca2241.png)

If the focus isn't on the editor window, but on the tree-view the terminal will always `cd` to the Design-folder's path - regardless which subfolder I have selected:

![screenshot 2016-05-16 14 12 25](https://cloud.githubusercontent.com/assets/11627131/15289117/3b21b0c0-1b70-11e6-8e15-9570f150638b.png)

It would be a nice addition if you could select a certain folder in your tree-view and `cd`directly to it, instead of going to the root.

Best wishes
Robert

**Comment by tkw1536** [comment]
Hi @rm-code there are actually 2 different commands in atomerinal:
1. `atomerinal:open-terminal-here` which looks at the current file in the editor, finds its path and opens a terminal there. 
2. `atomerinal:open-terminal-in-root` which currently just opens a terminal in **each** root folder (see the code https://github.com/tkw1536/atomerinal/blob/master/lib/atomerinal.coffee#L34). 

If I understand you right you want 2 new things: 
1. `atomerinal:open-terminal-in-root` should open the root folder associated with the selected tree view entry
2. a new command that opens the current folder that is selected in tree view

I can look into the tree-view API to see how difficult this is and will do so once I have a bit more time (probably next week).

**Comment by rm-code** [comment]
I did some further tests so let me try to clarify:

The issue I was talking about only happens with `atomerinal:open-terminal-in-root`. When I have two root folders (e.g.: `love` and `design`) Atomerinal will always open the Terminal in both folders which means I end up with two open terminal windows. Since `design` is opened later it will always show up on top.

> 1. atomerinal:open-terminal-in-root should open the root folder associated with the selected tree view entry

This should do the trick I guess. I am not sure if you could check which of the folders the selected file is located in.

> 1. a new command that opens the current folder that is selected in tree view

Yes, that's basically what I had in mind (should I open a different ticket for this?). Ideally `atomerinal:open-terminal-here` would check if the tree-view has focus and then try to open the selected folder.

Hope this helps 😅

**Comment by tkw1536** [comment]
Thanks for the clarification. Since these two are very closely related we can keep them in the same issue for now.

**Comment by tkw1536** [comment]
@rm-code I have implemented both of the features mentioned above: 
1. atomerinal:open-terminal-in-root now finds the currently associated project root
2. atomerinal:open-terminal-here now checks if the tree-view has focus and if so opens a file based on that

I have also added two context menu items to tree-view that check the currently selected tree-view item and open a terminal relative to that. 

I have only tested this on Windows so far. I will check if it works on Linux and Mac tomorrow and then make a proper release on the package registry.
